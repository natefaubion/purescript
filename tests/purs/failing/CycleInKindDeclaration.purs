-- @shouldFailWith CycleInKindDeclaration
module Main where

data Foo :: Bar -> Type
data Foo a = Foo

data Bar :: Foo -> Type
data Bar a = Bar
