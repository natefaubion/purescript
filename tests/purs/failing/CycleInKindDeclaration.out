Error found:
at tests/purs/failing/CycleInKindDeclaration.purs:7:1 - 7:24 (line 7, column 1 - line 7, column 24)

  A cycle appears in a set of kind declarations:

    {[33mBar[0m, [33mFoo[0m}

  Kind declarations may not refer to themselves in their own signatures.


See https://github.com/purescript/documentation/blob/master/errors/CycleInKindDeclaration.md for more information,
or to contribute content related to this error.

